functions/functionsv2/label_gce_instance/label_gce_instance.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_label_gce_instance]
+
+// Package helloworld provides a set of Cloud Functions samples.
+package helloworld
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"github.com/cloudevents/sdk-go/v2/event"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+// AuditLogEntry represents a LogEntry as described at
+// https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry
+type AuditLogEntry struct {
+	ProtoPayload *AuditLogProtoPayload `json:"protoPayload"`
+}
+
+// AuditLogProtoPayload represents AuditLog within the LogEntry.protoPayload
+// See https://cloud.google.com/logging/docs/reference/audit/auditlog/rest/Shared.Types/AuditLog
+type AuditLogProtoPayload struct {
+	MethodName         string                 `json:"methodName"`
+	ResourceName       string                 `json:"resourceName"`
+	AuthenticationInfo map[string]interface{} `json:"authenticationInfo"`
+}
+
+var client *compute.InstancesClient
+
+func init() {
+	// Create an Instances Client
+	var err error
+	client, err = compute.NewInstancesRESTClient(context.Background())
+	if err != nil {
+		log.Fatalf("Failed to create instances client: %s", err)
+	}
+}
+
+// Cloud Function that receives GCE instance creation Audit Logs, and adds a
+// `creator` label to the instance.
+func LabelGceInstance(ctx context.Context, ev event.Event) error {
+	// Extract parameters from the Cloud Event and Cloud Audit Log data
+	logentry := &AuditLogEntry{}
+	if err := ev.DataAs(logentry); err != nil {
+		werr := fmt.Errorf("event.DataAs() : %w", err)
+		log.Printf("Error parsing proto payload: %s", werr)
```

Comment:
nit: I am not sure we need these extra logs where errors are returned. Seem duplicative.

Reply:
That was my initial though as well, but in my experience, just returning the error gets you a log message that the response was a 500, but does not include the string representation of the returned error.
I think the added logging is worthwhile, since it provides necessary debugging info for users of the sample.

Reply:
Ack, we can do it here if you feel strongly, but this is not something we have been doing in general with our samples.